test/integration/targets/docker_node/tasks/test_node.yml

```diff
@@ -0,0 +1,856 @@
+         - 'output.swarm_facts.JoinTokens.Worker'
+
+  - name: Try to get docker_node_facts when docker is running in swarm mode and as manager
+    docker_node_facts:
+    register: output
+
+  - name: assert reading docker swarm node facts
+    assert:
+      that:
+         - 'output.nodes_facts | length > 0'
+         - 'output.nodes_facts[0].ID is string'
+
+  - name: Register node ID
+    set_fact:
+      nodeid: "{{ output.nodes_facts[0].ID }}"
+
+####################################################################
+## Set node as swarm manager #######################################
+####################################################################
+
+  - name: Try to set node as manager (check)
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: manager
+    check_mode: yes
+    register: set_as_manager_1
+
+  - name: Try to set node as manager
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: manager
+    register: set_as_manager_2
+
+  - name: Try to set node as manager (idempotent)
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: manager
+    register: set_as_manager_3
+
+  - name: Try to set node as manager (idempotent check)
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: manager
+    check_mode: yes
+    register: set_as_manager_4
+
+  - name: assert that node role does has not changed
+    assert:
+      that:
+         - 'set_as_manager_1 is not changed'
+         - 'set_as_manager_2 is not changed'
+         - 'set_as_manager_3 is not changed'
+         - 'set_as_manager_4 is not changed'
+         - 'set_as_manager_1.node_facts.Spec.Role == "manager"'
+         - 'set_as_manager_2.node_facts.Spec.Role == "manager"'
+         - 'set_as_manager_3.node_facts.Spec.Role == "manager"'
+         - 'set_as_manager_4.node_facts.Spec.Role == "manager"'
+
+####################################################################
+## Set node as swarm worker ########################################
+####################################################################
+
+  - name: Try to set node as worker (check)
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: worker
+    check_mode: yes
+    register: set_as_worker_1
+
+  - name: Try to set node as worker
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: worker
+    ignore_errors: yes
+    register: set_as_worker_2
+
+  - name: Try to set node as worker (idempotent)
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: worker
+    ignore_errors: yes
+    register: set_as_worker_3
+
+  - name: Try to set node as worker (idempotent check)
+    docker_node:
+      hostname: "{{ nodeid }}"
+      role: worker
+    check_mode: yes
+    register: set_as_worker_4
+
+  - name: assert that node cannot change role to worker
+    assert:
+      that:
+         - 'set_as_worker_1 is changed'
+         - 'set_as_worker_2 is failed'
+         - 'set_as_worker_2 is failed'
```

Comment:
I think this should be `_3` (and the next `_4`)

Reply:
Anyway, here you don't really need idempotency and idempotency+check mode, since they do the same as the first two checks. Idempotency and idempotency+check is only needed when the regular task actually changes something (here it's just failing, and shouldn't change anything).

Reply:
Well, it shouldn't but isn't the test to exclude human error if it actually changes something?